src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/publication/FileGenerationTest.java

```diff
@@ -0,0 +1,271 @@
+
+    @Value("${test-user-id}")
+    public String testUserId;
+
+    @BeforeAll
+    public void setup() {
+        doPostRequest(
+            TESTING_SUPPORT_LOCATION_URL + COURT_ID,
+            Map.of(AUTHORIZATION, BEARER + accessToken), COURT_NAME
+        );
+    }
+
+    @BeforeEach
+    public void setupEach() {
+        headerMap = new ConcurrentHashMap<>();
+        headerMap.put(AUTHORIZATION, BEARER + accessToken);
+    }
+
+    @AfterAll
+    public void teardown() {
+        doDeleteRequest(TESTING_SUPPORT_PUBLICATION_URL + COURT_NAME, Map.of(AUTHORIZATION, BEARER + accessToken));
+        doDeleteRequest(TESTING_SUPPORT_LOCATION_URL + COURT_NAME, Map.of(AUTHORIZATION, BEARER + accessToken));
+    }
+
+    private Artefact uploadPublication(
+        ListType listType, String file, Language language, Sensitivity sensitivity) throws IOException {
+        Map<String, String> headerMap = new ConcurrentHashMap<>();
+        headerMap.put(AUTHORIZATION, BEARER + accessToken);
+        headerMap.put(PublicationConfiguration.TYPE_HEADER, ARTEFACT_TYPE.toString());
+        headerMap.put(PublicationConfiguration.PROVENANCE_HEADER, PROVENANCE);
+        headerMap.put(PublicationConfiguration.SOURCE_ARTEFACT_ID_HEADER, SOURCE_ARTEFACT_ID);
+        headerMap.put(PublicationConfiguration.DISPLAY_FROM_HEADER, DISPLAY_FROM.toString());
+        headerMap.put(PublicationConfiguration.DISPLAY_TO_HEADER, DISPLAY_FROM.plusDays(1).toString());
+        headerMap.put(PublicationConfiguration.COURT_ID, COURT_ID);
+        headerMap.put(PublicationConfiguration.LIST_TYPE, listType.toString());
+        headerMap.put(PublicationConfiguration.CONTENT_DATE, contentDate.toString());
+        headerMap.put(PublicationConfiguration.SENSITIVITY_HEADER, sensitivity.toString());
+        headerMap.put(PublicationConfiguration.LANGUAGE_HEADER, language.toString());
+        headerMap.put("Content-Type", "application/json");
+
+        Artefact artefact;
+        try (InputStream jsonFile = this.getClass().getClassLoader()
+            .getResourceAsStream(file)) {
+            final String jsonString = new String(jsonFile.readAllBytes(), StandardCharsets.UTF_8);
+
+            final Response response = doPostRequest(
+                UPLOAD_PUBLICATION_URL,
+                headerMap, jsonString
+            );
+            assertThat(response.getStatusCode()).isEqualTo(CREATED.value());
+            contentDate = contentDate.plusDays(1);
+            artefact = response.getBody().as(Artefact.class);
+        }
+
+        Awaitility.with().pollInterval(1, TimeUnit.SECONDS).await().until(() -> {
+            Response existsResponse = doGetRequest(String.format(FILE_EXISTS_URL, artefact.getArtefactId()), headerMap);
+            return existsResponse.getStatusCode() == OK.value() && existsResponse.getBody().as(Boolean.class);
+        });
+
+        Response existsResponse = doGetRequest(String.format(FILE_EXISTS_URL, artefact.getArtefactId()), headerMap);
```

Comment:
This bit here looks a bit odd having to call another GET outside of the previous awaitility call. If awaitility timeout occurs and no file is returned, I assume an exception will be thrown. I am thinking wrapping the awaitility call with asssertDoesNotThrow so there is no need to do the GET endpoint again?